internal/engine/ingester/deps/deps.go

```diff
@@ -0,0 +1,225 @@
+	var logger = zerolog.Ctx(ctx)
+	userCfg := &Config{
+		Branch: defaultBranch,
+	}
+	if err := mapstructure.Decode(params, userCfg); err != nil {
+		return nil, fmt.Errorf("failed to read dependency ingester configuration from params: %w", err)
+	}
+
+	if repo.GetCloneUrl() == "" {
+		return nil, fmt.Errorf("could not get clone url")
+	}
+
+	branch := gi.getBranch(repo, userCfg.Branch)
+	logger.Info().Interface("repo", repo).Msgf("extracting dependencies from %s#%s", repo.GetCloneUrl(), branch)
+
+	// We clone to the memfs go-billy filesystem driver, which doesn't
+	// allow for direct access to the underlying filesystem. This is
+	// because we want to be able to run this in a sandboxed environment
+	// where we don't have access to the underlying filesystem.
+	r, err := gi.gitprov.Clone(ctx, repo.GetCloneUrl(), branch)
+	if err != nil {
+		if errors.Is(err, provifv1.ErrProviderGitBranchNotFound) {
+			return nil, fmt.Errorf("%w: %s: branch %s", engerrors.ErrEvaluationFailed,
+				provifv1.ErrProviderGitBranchNotFound, branch)
+		} else if errors.Is(err, provifv1.ErrRepositoryEmpty) {
+			return nil, fmt.Errorf("%w: %s", engerrors.ErrEvaluationSkipped, provifv1.ErrRepositoryEmpty)
+		}
+		return nil, err
+	}
+
+	wt, err := r.Worktree()
+	if err != nil {
+		return nil, fmt.Errorf("could not get worktree: %w", err)
+	}
+
+	deps, err := scanFs(ctx, wt.Filesystem)
+	if err != nil {
+		return nil, fmt.Errorf("could not scan filesystem: %w", err)
+	}
+
+	logger.Debug().Interface("deps", deps).Msgf("Scanning successful: %d nodes found", len(deps.Nodes))
+
+	head, err := r.Head()
+	if err != nil {
+		return nil, fmt.Errorf("could not get head: %w", err)
+	}
+
+	hsh := head.Hash()
+
+	chkpoint := checkpoints.NewCheckpointV1Now().
+		WithBranch(branch).
+		WithCommitHash(hsh.String())
+
+	return &interfaces.Result{
+		Object: map[string]any{
+			"node_list": deps,
+		},
+		Checkpoint: chkpoint,
+	}, nil
+}
+
+func (gi *Deps) getBranch(repo *pb.Repository, branch string) string {
+	// If the user has specified a branch, use that
+	if branch != "" {
+		return branch
+	}
+
+	// If the branch is provided in the rule-type
+	// configuration, use that
+	if gi.cfg.GetRepo().Branch != "" {
+		return gi.cfg.GetRepo().Branch
+	}
+	if repo.GetDefaultBranch() != "" {
+		return repo.GetDefaultBranch()
+	}
+
+	// If the branch is not provided in the rule-type
+	// configuration, use the default branch
+	return defaultBranch
+}
+
+func scanFs(ctx context.Context, memFS billy.Filesystem) (*sbom.NodeList, error) {
+	if memFS == nil {
+		return nil, fmt.Errorf("unable to scan dependencies, no active defined")
+	}
+	// have to down-cast here, because scalibr needs multiple io/fs types
+	wrapped, ok := iofs.New(memFS).(scalibr_fs.FS)
+	if !ok {
+		return nil, fmt.Errorf("error converting filesystem to ReadDirFS")
+	}
+
+	desiredCaps := scalibr_plugin.Capabilities{
+		OS:            scalibr_plugin.OSLinux,
+		Network:       true,
```

Comment:
what do we need the network option for? We might wanna add a small comment for this.

Reply:
It's to select the plugins that require network access. I'll add a note.